loki/transformations/tests/test_data_offload.py

```diff
@@ -81,11 +117,25 @@ def test_data_offload_region_openacc(frontend, assume_deviceptr):
     end do
   END SUBROUTINE kernel_routine
 """
-    driver = Sourcefile.from_source(fcode_driver, frontend=frontend)['driver_routine']
-    kernel = Sourcefile.from_source(fcode_kernel, frontend=frontend)['kernel_routine']
-    driver.enrich(kernel)
-
-    driver.apply(DataOffloadTransformation(assume_deviceptr=assume_deviceptr), role='driver',
+    with caplog.at_level(log_levels['ERROR']):
```

Comment:
I'd keep the caplog context tightly within the "expected failure" conditional below.

Reply:
Ignore, I had misunderstood how caplog works. This is fine.